### ISSUE TITLE
cannot use intersphinx if Python installation has no SSL support

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#801]
i have a Python installation that is built without SSL support.  as a result, urllib2.HTTPSHandler cannot be imported when i try to use intersphinx.  a `make html` results in the following traceback:

```
# Sphinx version: 1.1                                                                 
# Python version: 2.7.1                                                               
# Docutils version: 0.8.1 release                                                     
# Jinja2 version: 2.6                                                                 
Traceback (most recent call last):                                                    
  File "/home/jira/pyenv/lib/python2.7/site-packages/sphinx/cmdline.py", line 188, in main                                                                                  
    warningiserror, tags)                                                             
  File "/home/jira/pyenv/lib/python2.7/site-packages/sphinx/application.py", line 114, in __init__                                                                          
    self.setup_extension(extension)                                                   
  File "/home/jira/pyenv/lib/python2.7/site-packages/sphinx/application.py", line 247, in setup_extension                                                                   
    mod = __import__(extension, None, None, ['setup'])
  File "/home/jira/pyenv/lib/python2.7/site-packages/sphinx/ext/intersphinx.py", line 44, in <module>
    handlers.append(urllib2.HTTPSHandler)
AttributeError: 'module' object has no attribute 'HTTPSHandler'
```

but if i change the line intersphinx.py:45 from

```
except NameError:
```

to

```
except AttributeError:
```

it works fine.  according to NameError's documentation, it would get raised only if an unqualified name was not found.  but 'handlers' and 'urllib2' are the only unqualified names on line 44.  thus the reference to 'urllib2.HTTPSHandler' raises an AttributeError.  since line 41 would have already raised a NameError if either 'handlers' or 'urllib2' was undefined, i think the author's intent was to catch only AttributeError.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/801
- Originally reported by: Ivan Vasin
- Originally created at: 2011-10-19T00:20:07.757

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-11-01 08:01:57+00:00_

Closes #801: Make intersphinx work properly without SSL support.

→ <<cset 5d19b860e2310213a536818518b53019b1dd6980>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:33:11+00:00_

Removing version: 1.1 (automated comment)
